tools/clang/test/CodeGenHLSL/quick-test/mad_opt.hlsl

```diff
@@ -0,0 +1,9 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
+
+
+// Make sure a is not used.
+// CHECK-NOT: dx.op.loadInput.f32(i32 4, i32 0
+
+float main(float a : A, float b :B) : SV_Target {
+  return mad(a, 0, b);
```

Comment:
(Note: I'm not related and/or in any position to accept this PR) could you add a test that does `mad(0, a, b)` too? Just to make sure both variants work as expected!